test/testunusedvar.cpp

```diff
@@ -6957,7 +6957,7 @@ class TestUnusedVar : public TestFixture {
                               "  int x = 0;\n"
                               "  std::for_each(ints.begin(), ints.end(), [&x](int i){ x += i; });\n"
                               "}");
-        TODO_ASSERT_EQUALS("[test.cpp:3]: (style) Variable 'x' is assigned a value that is never used.\n", "", errout.str());
+        ASSERT_EQUALS("[test.cpp:2]: (style) Variable 'x' is assigned a value that is never used.\n", errout.str());
```

Comment:
is it possible that it thinks that "std::for_each" is a function declaration.. I think that in the executable scopes we should only treat it as a function declaration if we see a "extern" or something like that.

Reply:
Matching only `%name% (` won't be enough. Clearly `::` is not a return type, but I'm not sure if relying on the presence of `extern` would help much.

Reply:
check `%name% (` is just the prerequisite to enter the check `isFunction`. As `isFunction` requires extra arguments to be created. `isFunction()` Checks if it's indeed a function and than we will add it but only if it's an `extern` and is not yet in the base.

> Edit: but maybe `isFunction()` already handles that?

yes it does lines `symboldatabase.cpp:1971-72`